### ISSUE TITLE
Maximum number of AgentPools in Kubernetes

### ISSUE THREAD

**Issue by javierprovecho** [Azure/acs-engine#3325]
**Is this a request for help?**:

No
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):

ISSUE
---

**What version of acs-engine?**:

v0.16.2
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**

Kubernetes 1.9

**What happened**:

```json
{
    "error": {
        "code": "InvalidTemplate",
        "message": "Deployment template validation failed: 'The number of template variables limit exceeded. Limit: '256' and actual: '260'. Please see https://aka.ms/arm-template/#variables for usage details.'."
    }
}
```

**What you expected to happen**:

```
ERROR: The maximum number of agentpools supported is 5, you are defining "6"
```

**How to reproduce it** (as minimally and precisely as possible):

run `acs-engine deploy` using an `apimodel.json` with 6 agentpools and 3 masters

**Anything else we need to know**:

1. The first issue should be the error message, I would prefer that acs-engine validate my `apimodel.json` looking if I'm passing the maximum number of agentpools
2. For v0.16.x there are 2 situations:
    - When defining 1 master and 6 agentpools, deployment pass with a top of 255 vars used
    - When defining 3 masters and 6 agentpools, deployment fails with a top of 260 vars used (as said above). The difference is located on the vars used for defining Kubernetes Internal LB.
3. We tested also v0.18.x and found that the maximum number of agentpools with 3 masters reduces from 5 to 4, again because new vars have been added to the ARM template like `singlequote` and `acsversion`. In general lines, I think this behavior may breaks updates from previous versions in clusters with more agentpools than supported from a newer version.

**Comment by serbrech** [comment]
I started implementing a fix for this in this PR : #2810, but I have been unable to dedicate time to this

**Comment by javierprovecho** [comment]
Thanks @serbrech , I searched for this kind of issue before posting but definitely didn't use the correct keywords. I'm going to reference here this issue (#2553) too.

**Comment by jackfrancis** [comment]
Fixed with #3649
